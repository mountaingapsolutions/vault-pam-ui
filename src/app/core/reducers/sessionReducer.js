import sessionAction from 'app/core/actions/sessionAction';

/**
 * Reducer class responsible for reacting to and handling the session REST calls.
 *
 */

/**
 * The primary reduce method.
 *
 * @public
 * @param {Object} previousState - The previous state.
 * @param {Object} action - Action type.
 * @returns {Object} The updated state.
 */
export default (previousState = {
    authUser: {},
    vaultDomain: {},
    vaultLookupSelf: {},
    sealStatus: {},
    vaultToken: {}
}, action) => {
    switch (action.type) {
        case sessionAction.ACTION_TYPES.AUTHENTICATE_USER_PASS:
            // Message the error message with proper grammar.
            if (action.errors) {
                action.errors = action.errors.map(err => `Authentication failed: ${err}.`);
            }
            return {...previousState, authUser: action};
        case sessionAction.ACTION_TYPES.SET_DOMAIN:
            return {...previousState, vaultDomain: action};
        case sessionAction.ACTION_TYPES.SET_TOKEN:
            return {...previousState, vaultToken: action};
        case sessionAction.ACTION_TYPES.VALIDATE_DOMAIN:
            return {...previousState, sealStatus: action};
        case sessionAction.ACTION_TYPES.VALIDATE_TOKEN:
            // Message the error message with proper grammar.
            if (action.errors) {
                action.errors = action.errors.map(err => `Authentication failed: ${err}.`);
            }
            return {...previousState, vaultLookupSelf: action};
        default:
            return {...previousState};
    }
};
