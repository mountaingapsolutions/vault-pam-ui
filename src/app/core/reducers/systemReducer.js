import systemAction from 'app/core/actions/systemAction';

/**
 * Reducer class responsible for reacting to and handling the system REST calls.
 *
 */

/**
 * The primary reduce method.
 *
 * @public
 * @param {Object} previousState - The previous state.
 * @param {Object} action - Action type.
 * @returns {Object} The updated state.
 */
export default (previousState = {
    config: {},
    selfCapabilities: {},
    sealStatus: {}
}, action) => {
    switch (action.type) {
        case systemAction.ACTION_TYPES.GET_CONFIG:
            return {...previousState, config: action.data || {}, action};
        case systemAction.ACTION_TYPES.GET_SELF_CAPABILITIES:
            return {...previousState, selfCapabilities: action.data || {}, action};
        case systemAction.ACTION_TYPES.GET_SEAL_STATUS:
            return {...previousState, sealStatus: action.data || {}, action};
        default:
            return {...previousState};
    }
};
