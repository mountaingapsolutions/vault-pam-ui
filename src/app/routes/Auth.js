import {withStyles} from '@material-ui/core/styles';
import {AppBar, Card, Grid, Toolbar, Typography} from '@material-ui/core';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {Redirect, Route, Switch, withRouter} from 'react-router-dom';

import Server from 'app/routes/auth/Server';
import Login from 'app/routes/auth/Login';
import Constants from 'app/util/Constants';

/**
 * The authentication container.
 */
class Auth extends Component {

    /**
     * Required React Component lifecycle method. Returns a tree of React components that will render to HTML.
     *
     * @override
     * @protected
     * @returns {ReactElement}
     */
    render() {
        const {classes, history, location, vaultDomain} = this.props;
        // If attempting to go to the login page with no domain set, go back to the initial set domain page.
        if (location.pathname === '/auth/login' && !vaultDomain.data) {
            history.push('/auth/server');
        }
        return <div>
            <AppBar position='static'>
                <Toolbar>
                    <img alt='logo' className='mr-1' src='/assets/vault-dark.svg'/>
                    <Typography noWrap color='inherit' variant='h6'>
                        {Constants.APP_TITLE}
                    </Typography>
                </Toolbar>
            </AppBar>
            <Grid container className='mt-1' justify='center'>
                <Card className={classes.card}>
                    <Switch>
                        <Redirect exact from='/' to='/auth/server'/>
                        <Route exact component={Server} path='/auth/server'/>
                        <Route exact component={Login} path='/auth/login'/>
                        <Redirect to='/auth/server'/>
                    </Switch>
                </Card>
            </Grid>
        </div>;
    }
}

Auth.propTypes = {
    classes: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    vaultDomain: PropTypes.object.isRequired
};

/**
 * Returns the Redux store's state that is relevant to this class as props.
 *
 * @private
 * @param {Object} state - The initial state.
 * @returns {Object}
 */
const _mapStateToProps = (state) => {
    return {
        ...state.sessionReducer
    };
};

/**
 * Returns custom style overrides.
 *
 * @private
 * @returns {Object}
 */
const _styles = () => ({
    card: {
        width: '800px'
    }
});

export default withRouter(connect(_mapStateToProps)(withStyles(_styles)(Auth)));
