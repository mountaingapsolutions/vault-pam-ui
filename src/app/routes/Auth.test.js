/* global expect, it, jest */
import {createMemoryHistory} from 'history';
import React from 'react';
import renderer from 'react-test-renderer';
import {applyMiddleware, combineReducers, createStore} from 'redux';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';

import sessionReducer from 'app/core/reducers/sessionReducer';
import systemReducer from 'app/core/reducers/systemReducer';

import Auth from './Auth';
import reduxThunk from 'redux-thunk';

jest.mock('app/routes/auth/Login');

/**
 * Configures the application store by invoking Redux's createStore method.
 *
 * @private
 * @param {Object} [initialState] = The initial state.
 * @returns {Object}
 */
const _configureStore = (initialState) => {
    return createStore(
        combineReducers({
            sessionReducer,
            systemReducer
        }),
        initialState,
        applyMiddleware(reduxThunk)
    );
};

/**
 * Renders an instance of Auth for testing.
 *
 * @private
 * @returns {ReactElement}
 */
const _getInstance = () => {
    return <Provider store={_configureStore()}>
        <Router history={createMemoryHistory()}>
            <Auth/>
        </Router>
    </Provider>;
};

it('renders correctly', () => {
    expect(renderer.create(_getInstance()).toJSON()).toMatchSnapshot();
});
