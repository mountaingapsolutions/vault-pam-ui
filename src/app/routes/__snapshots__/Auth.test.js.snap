// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <img
        alt="logo"
        className="mr-1"
        src="/assets/vault-dark.svg"
      />
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Vault PAM UI
      </h6>
    </div>
  </header>
  <div
    className="MuiGrid-root mt-1 MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root Auth-card-1"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiCardContent-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Sign in to Vault
          </h6>
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <header
              className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault"
            >
              <div
                className="MuiTabs-root"
              >
                <div
                  className="MuiTabs-flexContainer"
                >
                  <div
                    className="MuiTabs-scroller MuiTabs-fixed"
                    onScroll={[Function]}
                    role="tablist"
                    style={
                      Object {
                        "marginBottom": null,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <div
                      className="MuiTabs-flexContainer"
                    >
                      <button
                        aria-selected={true}
                        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="tab"
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiTab-wrapper"
                        >
                          Token
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-selected={false}
                        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="tab"
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiTab-wrapper"
                        >
                          Userpass
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-selected={false}
                        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="tab"
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiTab-wrapper"
                        >
                          LDAP
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-selected={false}
                        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="tab"
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiTab-wrapper"
                        >
                          GitHub
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-selected={false}
                        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="tab"
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiTab-wrapper"
                        >
                          Okta
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <span
                      className="PrivateTabIndicator-root-205 PrivateTabIndicator-colorSecondary-207 MuiTabs-indicator"
                      style={
                        Object {
                          "left": 0,
                          "width": 0,
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </header>
            <fieldset
              className="MuiFormControl-root Login-formContainer-178"
            >
              <legend
                className="MuiFormLabel-root Login-formLabel-180"
              />
              <div
                className="Login-contentContainer-177"
              >
                <div
                  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root m-1"
                >
                  <label
                    className="MuiFormLabel-root Mui-required Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink={false}
                  >
                    Token
                    <span
                      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                      â€‰
                      *
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-261 MuiOutlinedInput-notchedOutline"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="PrivateNotchedOutline-legend-262"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      name="token"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing Login-cardAction-176"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-text MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Back
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-containedPrimary MuiButton-contained MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Next
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="Footer-footerRootContainer-283"
        >
          <header
            className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed Footer-footer-282"
          >
            <div
              className="Footer-footerTextContainer-284"
            >
              <a
                className="MuiTypography-root MuiTypography-caption MuiTypography-displayInline Footer-text-286"
                href="/rest/api"
                title="API"
              >
                API
              </a>
              <span
                className="MuiTypography-root MuiTypography-caption MuiTypography-displayInline Footer-textSeparator-285"
              >
                |
              </span>
              <a
                className="MuiTypography-root MuiTypography-caption MuiTypography-displayInline Footer-text-286"
                href=""
                title="Vault Domain"
              >
                
              </a>
              <span
                className="MuiTypography-root MuiTypography-caption MuiTypography-displayInline Footer-textSeparator-285"
              >
                |
              </span>
              <span
                className="MuiTypography-root MuiTypography-caption MuiTypography-displayInline Footer-text-286"
              >
                Vault 
              </span>
            </div>
          </header>
        </div>
      </form>
    </div>
  </div>
</div>
`;
